File: frontend/src/app/models/chat.models.ts
export enum ModelProvider {
  DeepSeekAPI = 0,
  LocalModel = 1
}

export interface ChatRequest {
  message: string;
  conversationId?: string;
  provider: ModelProvider;
}

export interface ChatResponse {
  response: string;
  conversationId: string;
  success: boolean;
  error?: string;
  provider: ModelProvider;
}

export interface ChatMessage {
  role: string; // 'user' or 'assistant'
  content: string;
  timestamp: Date;
}

export interface ProviderInfo {
  name: string;
  value: ModelProvider;
  available: boolean;
} 

File: backend/Models/ChatModels.cs
namespace DeepSeekChatApi.Models
{
    public enum ModelProvider
    {
        DeepSeekAPI,
        LocalModel
    }

    public class ChatRequest
    {
        public string Message { get; set; } = string.Empty;
        public string? ConversationId { get; set; }
        public ModelProvider Provider { get; set; } = ModelProvider.DeepSeekAPI;
    }

    public class ChatResponse
    {
        public string Response { get; set; } = string.Empty;
        public string ConversationId { get; set; } = string.Empty;
        public bool Success { get; set; }
        public string? Error { get; set; }
        public ModelProvider Provider { get; set; }
    }

    public class ChatMessage
    {
        public string Role { get; set; } = string.Empty; // "user" or "assistant"
        public string Content { get; set; } = string.Empty;
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    }
} 